lib/style-observer.js

```diff
@@ -0,0 +1,309 @@
+    var getSelector = function (filter) {
+        // getSelector will be triggered via mutation observer callbacks
+        // and we assume that those are already throttled.
+        examineStylesScheduler.runImmediately();
+        invalidateScheduler.run();
+
+        if (getSelectorCache[filter]) {
+            return getSelectorCache[filter];
+        }
+        var styleSheets = document.styleSheets;
+        var selectors = [];
+        for (var l = styleSheets.length - 1; l >= 0; l--) {
+            var styleSheet = styleSheets[l];
+            if (styleSheet.disabled) { continue; }
+            var map = sheetToFilterSelectorMap.get(styleSheet);
+            if (typeof map === 'undefined') {
+                // This can happen with cross-origin styles.
+                continue;
+            }
+            Array.prototype.push.apply(selectors, map[filter]);
+        }
+        getSelectorCache[filter] = selectors;
+        return selectors;
+    };
+
+    var parsedFilters = [];
+
+    window.parsedFilters = parsedFilters;
+
+    var reRegexRule = /^\/(.*)\/$/;
+    var reProperty = /[A-Za-z0-9\-]+/;
+    var registerStylePropertyFilter = function (filter) {
+        filter = filter.trim();
+        var property, re;
+        if (reRegexRule.test(filter)) {
+            filter = filter.slice(1, -1);
+            var index = filter.indexOf(':');
+            property = filter.slice(0, index).trim();
+            if (!reProperty.test(property)) {
+                console.warn('Unsupported regex usage of :properties(' + filter + ')');
+                return;
+            }
+            re = utils.toRegex(filter.slice(index + 1));
+        } else {
+            var index = filter.indexOf(':');
+            property = filter.slice(0, index).trim();
+            var simple = filter.slice(index + 1).trim();
+            re = new RegExp(utils.createSimpleRegex(simple), 'i');
+        }
+        parsedFilters.push({
+            filter: filter,
+            property: property,
+            re: re
+        });
+    };
+
+    /**/
+
+    var initialized = false;
+    var initialize = function () {
+        if (initialized) { return; }
+        observeStyle();
+        var sheets = document.styleSheets;
+        for (var l = sheets.length - 1; l >= 0; l--) {
+            readStyleSheetContent(sheets[l]);
+        }
+    };
+
+    var extendSizzle = function(Sizzle) {
+        Sizzle.selectors.pseudos["properties"] = Sizzle.selectors.pseudos["-abp-properties"] = Sizzle.selectors.createPseudo(function(propertyFilter) {
+            registerStylePropertyFilter(propertyFilter);
+            return function (element) {
+                var selectors = getSelector(propertyFilter);
+                return element.matches(selectors.join(','));
+            }
+        });
+    };
+
+    return {
+        addIgnoredSheets: addIgnoredSheets,
+        initialize: initialize,
+        extendSizzle: extendSizzle,
+        getSelector: getSelector
```

Comment:
Why is `getSelector` exposed? I don't see where it's used externally.

Reply:
For debugging purposes